### PR TITLE
feat: replace custom sysctl implementation with stdlib

### PR DESCRIPTION
**Pull request by kruskall**
Remove custom string->mib conversion code and sysctl implementation.
Reduce code complexity and cgo dependency

Related to https://github.com/elastic/apm-server/issues/8718

### CODE REVIEW

File: providers/darwin/syscall_darwin.go [hunk 1]
```diff
@@ -33,116 +33,16 @@ import (
...
-	// kernel uses +2 for its own implementation of this function.
-	// I am scared that if we don't include the +2 here, the kernel
-	// will silently write 2 words farther than we specify
-	// and we'll get memory corruption.
-	var buf [C.CTL_MAXNAME + 2]C.int
-	n := uintptr(C.CTL_MAXNAME) * siz
-
-	p := (*byte)(unsafe.Pointer(&buf[0]))
-	bytes, err := syscall.ByteSliceFromString(name)
+func sysctl(name string, value interface{}, args ...int) error {
```
Comment (andrewkroh) [inline providers/darwin/syscall_darwin.go:41]: I think we should remove the `sysctl` and `sysctlByName` wrappers too and replace usages with direct calls into `unix.SysctlRaw`.

**Review by andrewkroh — COMMENTED** [review]
The one concern I had was with performance. The only difference is that the unix package makes two `sysctl` calls (one to determine buf size and one to get the data). The old implementation in this package made one call with a max size buffer.

This does make it a bit slower, but it's not significant.

```go
func BenchmarkKernProcargs(b *testing.B) {
	pid := os.Getpid()
	var p process
	for i := 0; i < b.N; i++ {
		kern_procargs(pid, &p)
	}
}
```

```
% benchcmp before-self.txt after-self.txt                                                      
benchcmp is deprecated in favor of benchstat: https://pkg.go.dev/golang.org/x/perf/cmd/benchstat
benchmark                    old ns/op     new ns/op     delta
BenchmarkKernProcargs-10     13783         15075         +9.37%

benchmark                    old allocs     new allocs     delta
BenchmarkKernProcargs-10     106            109            +2.83%

benchmark                    old bytes     new bytes     delta
BenchmarkKernProcargs-10     31931         34104         +6.81%
```

```
% benchstat before-self.txt after-self.txt
name             old time/op    new time/op    delta
KernProcargs-10    13.8µs ± 0%    15.1µs ± 0%   ~     (p=1.000 n=1+1)

name             old alloc/op   new alloc/op   delta
KernProcargs-10    31.9kB ± 0%    34.1kB ± 0%   ~     (p=1.000 n=1+1)

name             old allocs/op  new allocs/op  delta
KernProcargs-10       106 ± 0%       109 ± 0%   ~     (p=1.000 n=1+1)
```

**Comment by andrewkroh** [comment]
And can you please add changelog entries for this and the other recent changes.

File: providers/darwin/syscall_darwin.go (hunk 1 above)
Comment (kruskall) [reply providers/darwin/syscall_darwin.go]: Good point! Updated!

**Comment by kruskall** [comment]
Removed unused methods and added the changelog entry 👍

**Review by andrewkroh — APPROVED** [review]

**Pull request merged**